statistics/handle/internal/cache/inner.go

```diff
@@ -0,0 +1,36 @@
+// Copyright 2023 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cache
+
+import "github.com/pingcap/tidb/statistics"
+
+// StatsCacheInner is the interface to manage the statsCache, it can be implemented by map, lru cache or other structures.
+type StatsCacheInner interface {
```

Comment:
Better to add comments for each method to make it easier to understand and safer to use. 
For example:`what's the different between GetByQuery and Get?`, `what's the unit of Cost(), B, KB, MB?`, `these returned *stats.Tables are read-only?`

Reply:
Done